Update DXUT port and add Effects11 (FX) port

walbourn opened the pull request:
This updates the existing **DXUT** port to the latest version (it was last updated to the September 2016 release). The upstream project was updated with it's own CMake project and package support to simplify this port.

Also added a new **Effects11** port for the *Effects system for Direct3D 11*. Again, the upstream project now provides a CMake.

walbourn committed:
Update dxut port, added effects11 port

walbourn committed:
DXUT does not support UWP

walbourn committed:
Fixed FX11 CMake to support UWP

walbourn committed:
DXUT supports arm64 desktop

walbourn committed:
Update baseline

walbourn committed:
Now that these build with CMake, no need to exclude staticcrt

walbourn commented:
Validated with all supported triplets:

```
vcpkg install dxut:x86-windows
vcpkg install dxut:x86-windows-static
vcpkg install dxut:x64-windows
vcpkg install dxut:x64-windows-static
vcpkg install dxut:arm-windows
vcpkg install dxut:arm64-windows
vcpkg install dxut:arm64-windows-static

vcpkg install effects11:x86-windows
vcpkg install effects11:x86-windows-static
vcpkg install effects11:x64-windows
vcpkg install effects11:x64-windows-static
vcpkg install effects11:arm-windows
vcpkg install effects11:arm64-windows
vcpkg install effects11:arm64-windows-static
vcpkg install effects11:x86-uwp
vcpkg install effects11:x64-uwp
vcpkg install effects11:arm-uwp
vcpkg install effects11:arm64-uwp
```

walbourn committed:
rev'd port version

walbourn committed:
Fix versions

ras0219-msft commented on versions/d-/dxut.json:
```diff
@@ -1,5 +1,15 @@
+      "version-string": "11.25",
+      "port-version": 1
+    },
+    {
+      "git-tree": "61e6d008d15a69e283b20aed662576c8636c3b73",
+      "version-string": "11.25",
+      "port-version": 0
+    },
```
```suggestion
```

Post-merge, this version won't be available due to rebasing.

ras0219-msft commented on versions/e-/effects11.json:
```diff
@@ -0,0 +1,14 @@
+      "git-tree": "62f8ad6b069eb4cf46a152d4a42ec2c883de0758",
+      "version-string": "11.26",
+      "port-version": 1
+    },
+    {
+      "git-tree": "aa02ef34a172ea1cf0d4b4b5c3c7aedb85133a05",
+      "version-string": "11.26",
+      "port-version": 0
+    }
```
```suggestion
    }
```
Post-merge, this version won't be available due to rebasing.

ras0219-msft reviewed (changes requested):
Awesome, thanks for the PR!

LGTM once my two suggestions are applied.

walbourn committed:
Update versions/d-/dxut.json

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

walbourn committed:
Update versions/e-/effects11.json

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

NancyLi1013 commented on ports/dxut/vcpkg.json:
```diff
@@ -1,8 +1,8 @@
-  "version-string": "11.14",
-  "port-version": 4,
+  "version-string": "11.25",
+  "port-version": 1,
```
```suggestion
  "port-version": 0,
```
Since this is new version update, it should be `0`.

NancyLi1013 commented on ports/effects11/vcpkg.json:
```diff
@@ -0,0 +1,8 @@
+{
+  "name": "effects11",
+  "version-string": "11.26",
+  "port-version": 1,
```
```suggestion
  "port-version": 0,
```
For a new port, this field also should be `0`.

NancyLi1013 commented on versions/baseline.json:
```diff
@@ -1689,8 +1689,8 @@
-      "baseline": "11.14",
-      "port-version": 4
+      "baseline": "11.25",
+      "port-version": 1
```
```suggestion
      "port-version": 0
```

NancyLi1013 commented on versions/baseline.json:
```diff
@@ -1736,6 +1736,10 @@
     },
+    "effects11": {
+      "baseline": "11.26",
+      "port-version": 1
```
```suggestion
      "port-version": 0
```

NancyLi1013 commented on versions/d-/dxut.json:
```diff
@@ -1,5 +1,10 @@
+    {
+      "git-tree": "44a8522987ce8c7e3582f30d98bff8a2a7b06136",
+      "version-string": "11.25",
+      "port-version": 1
```
```suggestion
      "port-version": 0
```

NancyLi1013 commented on versions/e-/effects11.json:
```diff
@@ -0,0 +1,9 @@
+    {
+      "git-tree": "62f8ad6b069eb4cf46a152d4a42ec2c883de0758",
+      "version-string": "11.26",
+      "port-version": 1
```
```suggestion
      "port-version": 0
```

walbourn replied:
I've done all that already. Maybe you need to pull?

walbourn replied:
I had a working 0 port in this PR. I can revert the 1 back manually as you suggest.

walbourn committed:
Update ports/dxut/vcpkg.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

walbourn committed:
Update versions/baseline.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

walbourn committed:
Update versions/d-/dxut.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

walbourn committed:
Update versions/e-/effects11.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

walbourn committed:
Update versions/baseline.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

walbourn committed:
Update ports/effects11/vcpkg.json

Co-authored-by: NancyLi1013 <46708020+NancyLi1013@users.noreply.github.com>

walbourn committed:
Updated version hash

NancyLi1013 commented:
Seems the format for vcpkg.json is not correct.

Could you please use `vcpkg format-manifest ./ports/xxx/vcpkg.json` to format it?

walbourn commented:
@NancyLi1013 It didn't like having a port-version of 0 in the vcpkg.json... Fixed.

walbourn committed:
Fixed format complaint

walbourn committed:
Fixed baseline

walbourn commented:
@NancyLi1013 So I've fixed all the issues with dxut and effects11, sync'd master, and rebased. 

The remaining failure isn't my port:

```
Error: While reading versions for port log4cpp from file: C:\a\1\s\versions\l-\log4cpp.json
        Version `2.9.1#4` was not found in versions file.
        Run:

           vcpkg x-add-version log4cpp

        to add the new port version.
```

walbourn committed:
Merge branch 'master' into dxutfx11

walbourn commented:
Merged !16138 so that should fix that failure...

walbourn commented:
@ras0219-msft @NancyLi1013 I believe at this point all your feedback has been integrated. Please approve.

ras0219-msft commented:
Yep, this is great! Thanks for the new versions and the update PR!
